app/views/markets/new.html.erb

```diff
@@ -0,0 +1,33 @@
+<%= form_for @market, url: { action: @action}, method: @method do |f| %>
```

Comment:
Since this is now being used for more than just the `new` operation, this should be renamed for clarity. I think you'd want to use the partial `_form` and then render that partial in both your `new.html.erb` file and the `edit.html.erb` file.